### PR TITLE
Db/fix collations

### PR DESCRIPTION
**Pull request by coffeegoddd**

### CODE REVIEW

File: sql/charsetcollation.go [hunk 1]
```diff
@@ -1273,3 +1294,49 @@ func (c Collation) WorksWithCharacterSet(cs CharacterSet) bool {
...
+	return s.ID
+}
+
+// IsDefault returns string specifying id collation is default.
+func (c Collation) IsDefault() string {
+	s, ok := CollationToMySQLVals[c]
+	if !ok {
+		return Y
+	}
+	return s.IsDefault
```
Comment (coffeegoddd) [inline sql/charsetcollation.go:1314]: should this be
```
if c == Collation_Default {
  return Y
}
return N
```
instead?

File: enginetest/queries.go [hunk 2]
```diff
@@ -1639,23 +1639,98 @@ var QueryTests = []QueryTest{
...
+				"binary",
+				sql.CollationToMySQLVals[sql.Collation_binary].ID,
+				sql.CollationToMySQLVals[sql.Collation_binary].IsDefault,
+				sql.CollationToMySQLVals[sql.Collation_binary].IsCompiled,
+				sql.CollationToMySQLVals[sql.Collation_binary].SortLen,
+				sql.CollationToMySQLVals[sql.Collation_binary].PadSpace,
+			},
+			{
+				sql.Collation_utf8_general_ci.String(),
+				"utf8mb3",
```
Comment (zachmu) [inline enginetest/queries.go:1654]: mysql reports this as "utf8"

**Review by zachmu — COMMENTED** [review]
Not quite what mysql does, see comments

File: sql/charsetcollation.go (hunk 1 above)
Comment (coffeegoddd) [reply sql/charsetcollation.go:1316]: leaving as is

File: enginetest/queries.go (hunk 2 above)
Comment (Hydrocharged) [reply enginetest/queries.go]: Darn it Zach, `utf8mb3` and `utf8` are aliases. `utf8` is deprecated so we're explicit in using `utf8mb3` if you pass in `utf8`.

**Pull request merged**